Run pep8 and ESLint on commit?

Me thinks we can use Travis-CI to do this, although not sure of the optimal way. I can try experimenting with this.

Was also looking at [Tox](https://tox.readthedocs.org/en/latest/) since I've seen that used in other projects, but not sure if that is overkill.

As for checking JavaScript files, we could use ESLint, but that may get tricky if we plan on excluding oodles of files/folders (in which case we may need Gulp or Grunt, and we're starting to add a few dependencies to our Travis env).

Probably best to just dive in and start doing a bunch of PRs and see what falls out.

For future reference, here's the initial fallout from a simple `git clone` and `pep8 .` command:

```
$ git clone git@github.com:mozilla-services/services-test.git
Cloning into 'services-test'...
remote: Counting objects: 1210, done.
remote: Total 1210 (delta 0), reused 0 (delta 0), pack-reused 1210
Receiving objects: 100% (1210/1210), 197.47 KiB | 0 bytes/s, done.
Resolving deltas: 100% (551/551), done.
Checking connectivity... done.

$ cd services-test

$ pep8 .
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:8:80: E501 line too long (80 > 79 characters)
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:16:12: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:17:1: W293 blank line contains whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:21:44: E261 at least two spaces before inline comment
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:21:46: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:22:78: E231 missing whitespace after ','
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:22:80: E501 line too long (81 > 79 characters)
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:23:55: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:24:80: E501 line too long (83 > 79 characters)
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:24:81: E231 missing whitespace after ','
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:25:80: E501 line too long (89 > 79 characters)
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:25:87: E231 missing whitespace after ','
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:26:50: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:28:66: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:29:66: W291 trailing whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:35:1: E302 expected 2 blank lines, found 1
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:37:1: W293 blank line contains whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:38:1: E302 expected 2 blank lines, found 1
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:39:1: W293 blank line contains whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:46:1: W293 blank line contains whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:49:1: E302 expected 2 blank lines, found 1
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:51:5: E265 block comment should start with '# '
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:56:1: W293 blank line contains whitespace
./autopush/e2e-test/tests/pop-notification.sikuli/pop-notification.py:90:11: W292 no newline at end of file
./kinto/api-test/test_kinto_buckets.py:24:80: E501 line too long (85 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:32:80: E501 line too long (93 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:44:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:75:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:84:80: E501 line too long (96 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:94:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:102:80: E501 line too long (83 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:109:80: E501 line too long (93 > 79 characters)
./kinto/api-test/test_kinto_buckets.py:121:80: E501 line too long (89 > 79 characters)
./kinto/api-test/test_kinto_collections.py:24:80: E501 line too long (85 > 79 characters)
./kinto/api-test/test_kinto_collections.py:32:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_collections.py:45:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_collections.py:61:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_collections.py:70:80: E501 line too long (96 > 79 characters)
./kinto/api-test/test_kinto_collections.py:78:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_collections.py:89:80: E501 line too long (82 > 79 characters)
./kinto/api-test/test_kinto_collections.py:106:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_collections.py:119:80: E501 line too long (89 > 79 characters)
./kinto/api-test/test_kinto_collections.py:133:80: E501 line too long (89 > 79 characters)
./kinto/api-test/test_kinto_collections.py:142:80: E501 line too long (105 > 79 characters)
./kinto/api-test/test_kinto_groups.py:25:80: E501 line too long (85 > 79 characters)
./kinto/api-test/test_kinto_groups.py:33:80: E501 line too long (92 > 79 characters)
./kinto/api-test/test_kinto_groups.py:45:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_groups.py:73:80: E501 line too long (92 > 79 characters)
./kinto/api-test/test_kinto_groups.py:92:80: E501 line too long (86 > 79 characters)
./kinto/api-test/test_kinto_groups.py:99:80: E501 line too long (105 > 79 characters)
./kinto/api-test/test_kinto_groups.py:121:80: E501 line too long (89 > 79 characters)
./kinto/api-test/test_kinto_heartbeat.py:9:80: E501 line too long (93 > 79 characters)
./kinto/api-test/test_kinto_records.py:93:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_records.py:113:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_records.py:133:80: E501 line too long (94 > 79 characters)
./kinto/api-test/test_kinto_records.py:166:80: E501 line too long (94 > 79 characters)
./msisdn-gateway/e2e-test/fabfile.py:4:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/fabfile.py:8:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:14:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:15:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:16:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:17:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:18:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:20:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:21:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:22:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:23:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:24:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:24:80: E501 line too long (102 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:25:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:26:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:28:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:29:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:29:80: E501 line too long (81 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:30:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:30:80: E501 line too long (106 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:31:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:32:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:33:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:33:80: E501 line too long (107 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:34:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:36:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:37:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:37:80: E501 line too long (90 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:38:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:39:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:40:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:40:80: E501 line too long (87 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:42:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:43:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:44:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:44:60: W291 trailing whitespace
./msisdn-gateway/e2e-test/test/control-script.py:45:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:47:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:48:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:48:26: W291 trailing whitespace
./msisdn-gateway/e2e-test/test/control-script.py:49:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:51:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:52:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:53:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:55:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:56:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:57:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:59:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:61:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:62:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:63:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:64:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:66:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:66:80: E501 line too long (83 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:67:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:68:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:68:80: E501 line too long (89 > 79 characters)
./msisdn-gateway/e2e-test/test/control-script.py:69:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:73:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:73:8: W292 no newline at end of file
./services-marionette/firefox-services-tests/apps/base.py:50:80: E501 line too long (80 > 79 characters)
./services-marionette/firefox-services-tests/apps/base.py:61:80: E501 line too long (80 > 79 characters)
./services-marionette/firefox-services-tests/apps/base.py:68:80: E501 line too long (80 > 79 characters)
./services-marionette/firefox-services-tests/apps/fxa/login.py:29:80: E501 line too long (98 > 79 characters)
./services-marionette/firefox-services-tests/apps/fxa/login.py:54:80: E501 line too long (83 > 79 characters)
./services-marionette/firefox-services-tests/apps/pocket/app.py:35:1: W391 blank line at end of file
```

Questions:
1. Do we care about pep8?
2. Do we care about all rules? Do we ignore "E501 line too long"?
3. Can we ignore certain folders from pep8 linting? We don't care about the "./autopush/e2e-test/tests/pop-notification.sikuli" folder since it's generated code.

Ignoring "E501" and the ".sikuli" folder, the output is reduced to this super manageable list:

```
$ pep8 . | grep -v ": E501 line too long" | grep -v ".sikuli"

./msisdn-gateway/e2e-test/fabfile.py:4:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/fabfile.py:8:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:14:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:15:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:16:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:17:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:18:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:20:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:21:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:22:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:23:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:24:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:25:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:26:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:28:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:29:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:30:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:31:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:32:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:33:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:34:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:36:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:37:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:38:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:39:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:40:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:42:1: E302 expected 2 blank lines, found 1
./msisdn-gateway/e2e-test/test/control-script.py:43:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:44:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:44:60: W291 trailing whitespace
./msisdn-gateway/e2e-test/test/control-script.py:45:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:47:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:48:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:48:26: W291 trailing whitespace
./msisdn-gateway/e2e-test/test/control-script.py:49:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:51:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:52:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:53:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:55:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:56:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:57:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:59:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:61:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:62:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:63:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:64:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:66:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:67:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:68:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:69:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:73:1: W191 indentation contains tabs
./msisdn-gateway/e2e-test/test/control-script.py:73:8: W292 no newline at end of file
./services-marionette/firefox-services-tests/apps/pocket/app.py:35:1: W391 blank line at end of file
```

Here's my ESLint config:

```
extends: eslint:recommended

env:
  es6: true
  node: true
  mocha: true

globals:
  marionette: true

rules:
  no-console: 0
  no-unused-vars: [2, {vars: all, args: none}]
```

And here's the current output:

```
$ eslint .

/Users/pdehaan/dev/tmp/del/services-test/dashboard/e2e-test/config/stage.js
  3:11  error  Unexpected trailing comma  comma-dangle

/Users/pdehaan/dev/tmp/del/services-test/dashboard/e2e-test/tests/dashboard.js
  6:5  error  "restmail" is defined but never used  no-unused-vars
  7:5  error  "P" is defined but never used         no-unused-vars

/Users/pdehaan/dev/tmp/del/services-test/loop/e2e-test/tests/loop_signup.js
  2:5  error  "expect" is defined but never used  no-unused-vars

✖ 4 problems (4 errors, 0 warnings)
```

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,0 +1,3 @@
+**/node_modules/*
+**/*_env/*
+**/venv/*

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,0 +1,14 @@
+extends: eslint:recommended
+
+env:
+  es6: true
+  node: true
+  mocha: true
+
+globals:
+  marionette: true
+  user_pref: true
+
+rules:
+  no-console: 0
+  no-unused-vars: [2, {vars: all, args: none}]

--- a/.flake8
+++ b/.flake8
@@ -1,0 +1,3 @@
+[flake8]
+ignore = E501
+exclude = **/*.sikuli/*,**/*_env/*,**/node_modules/*,**/venv/**

--- a/dashboard/e2e-test/config/stage.js
+++ b/dashboard/e2e-test/config/stage.js
@@ -1,4 +1,4 @@
 module.exports = {
   host: 'dashboard.stage.mozaws.net',
-  port: 80,
+  port: 80
 }

--- a/dashboard/e2e-test/tests/dashboard.js
+++ b/dashboard/e2e-test/tests/dashboard.js
@@ -3,9 +3,7 @@
 
 var chai = require('chai')
 var helper = require('marionette-helper')
-var restmail = require('restmail-client')
-var P = require('promise')
-var config = require('config');
+var config = require('config')
 
 var expect = chai.expect
 

--- a/kinto/api-test/test_kinto_records.py
+++ b/kinto/api-test/test_kinto_records.py
@@ -195,7 +195,7 @@ class Kinto_Records(unittest.TestCase):
         resource = 'buckets/test_bucket/collections/test_collection/records'
         expected_status_code = 405 if self.client.is_read_only() else 400
         data = '{"data": {"test": }'
-        response = self.client.post_request(
+        self.client.post_request(
             resource, data, status_code=expected_status_code)
 
     def test_utf8_stored_correctly(self):

--- a/loop/e2e-test/tests/loop_signup.js
+++ b/loop/e2e-test/tests/loop_signup.js
@@ -1,5 +1,4 @@
 var Promise = require('promise');
-var expect = require('chai').expect;
 
 var profile = require('../profile');
 

--- a/msisdn-gateway/e2e-test/fabfile.py
+++ b/msisdn-gateway/e2e-test/fabfile.py
@@ -1,9 +1,11 @@
-from fabric.api import *
-import fabric
+from fabric.api import *  # noqa
+# import fabric
+
 
 @task
 def setup():
     local("git clone https://github.com/kreamkorokke/msisdn-cli")
+
 
 @task
 def teardown():

--- a/msisdn-gateway/e2e-test/test/control-script.py
+++ b/msisdn-gateway/e2e-test/test/control-script.py
@@ -11,63 +11,66 @@ from twilio.rest import TwilioRestClient
 
 host = "https://msisdn.services.mozilla.com"
 
+
 def clean_up_message_logs(client):
-	# clean up the previous message logs
-	messages = client.messages.list()
-	for i in range(0, len(messages)):
-		client.messages.delete(messages[i].sid)
+    # clean up the previous message logs
+    messages = client.messages.list()
+    for i in range(0, len(messages)):
+        client.messages.delete(messages[i].sid)
+
 
 def run_msisdn_cli(client, cli_command, gateway_number, test_number):
-	# spawn a worker for msisdn-cli
-	worker = pexpect.spawn("bash")
-	worker.logfile = sys.stdout
-	worker.sendline("cd ./msisdn-cli; virtualenv .venv; . .venv/bin/activate; python ./setup.py develop")
-	worker.sendline(cli_command)
-	print("Executing command: " + cli_command)
+    # spawn a worker for msisdn-cli
+    worker = pexpect.spawn("bash")
+    worker.logfile = sys.stdout
+    worker.sendline("cd ./msisdn-cli; virtualenv .venv; . .venv/bin/activate; python ./setup.py develop")
+    worker.sendline(cli_command)
+    print("Executing command: " + cli_command)
 
-	# retrieve the verification message
-	worker.expect("Please\senter\sthe\scode\sthat\syou\swill\sget\sby\sSMS\sfrom\s")
-	message_log = client.messages.list(gateway_number, test_number, time.strftime("%Y-%m-%d", time.gmtime()))
-	while not message_log:
-		time.sleep(1)
-		message_log = client.messages.list(gateway_number, test_number, time.strftime("%Y-%m-%d", time.gmtime()))
-	verification_message = message_log[0].body
+    # retrieve the verification message
+    worker.expect("Please\senter\sthe\scode\sthat\syou\swill\sget\sby\sSMS\sfrom\s")
+    message_log = client.messages.list(gateway_number, test_number, time.strftime("%Y-%m-%d", time.gmtime()))
+    while not message_log:
+        time.sleep(1)
+        message_log = client.messages.list(gateway_number, test_number, time.strftime("%Y-%m-%d", time.gmtime()))
+    verification_message = message_log[0].body
 
-	# extract the code
-	match = re.search("^Your\sverification\scode\sis:\s(\d\d\d\d\d\d)", verification_message)
-	worker.sendline(match.group(1))
-	worker.expect("Verified")
-	print("++++++++++++++++++++ Number: " + test_number + " VERIFIED +++++++++++++++++++")
+    # extract the code
+    match = re.search("^Your\sverification\scode\sis:\s(\d\d\d\d\d\d)", verification_message)
+    worker.sendline(match.group(1))
+    worker.expect("Verified")
+    print("++++++++++++++++++++ Number: " + test_number + " VERIFIED +++++++++++++++++++")
+
 
 def main():
-	parser = argparse.ArgumentParser(
-		description='Scripts for automating msisdn-gateway test', 
-		formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser = argparse.ArgumentParser(
+        description='Scripts for automating msisdn-gateway test',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
 
-	parser.add_argument(
-		"-a", "--account-auth", 
-		required=True)
+    parser.add_argument(
+        "-a", "--account-auth",
+        required=True)
 
-	parser.add_argument(
-		"-n", "--numbers",
-		required=True)
+    parser.add_argument(
+        "-n", "--numbers",
+        required=True)
 
-	parser.add_argument(
-		"-H", "--host-number",
-		required=True)
+    parser.add_argument(
+        "-H", "--host-number",
+        required=True)
 
-	args = vars(parser.parse_args())
+    args = vars(parser.parse_args())
 
-	with open(args["account_auth"]) as account_auth_file:
-		account_auth = json.load(account_auth_file)
-	with open(args["numbers"]) as numbers_file:
-		numbers = json.load(numbers_file)
+    with open(args["account_auth"]) as account_auth_file:
+        account_auth = json.load(account_auth_file)
+    with open(args["numbers"]) as numbers_file:
+        numbers = json.load(numbers_file)
 
-	client = TwilioRestClient(account_auth["account_sid"], account_auth["auth_token"])
-	clean_up_message_logs(client)
-	command = "msisdn-cli -H " + host + " -c " + numbers["mcc"] + " -n " + numbers["number"]
-	run_msisdn_cli(client, command, args["host_number"], numbers["number"])
+    client = TwilioRestClient(account_auth["account_sid"], account_auth["auth_token"])
+    clean_up_message_logs(client)
+    command = "msisdn-cli -H " + host + " -c " + numbers["mcc"] + " -n " + numbers["number"]
+    run_msisdn_cli(client, command, args["host_number"], numbers["number"])
 
 
 if __name__ == "__main__":
-	main()
+    main()

--- a/package.json
+++ b/package.json
@@ -1,0 +1,18 @@
+{
+  "name": "services-test",
+  "description": "We test services",
+  "version": "1.0.0",
+  "author": "Cloud Services QA",
+  "dependencies": {},
+  "devDependencies": {
+    "eslint": "1.10.3"
+  },
+  "license": "MPL-2.0",
+  "private": true,
+  "repository": "mozilla-services/services-test",
+  "scripts": {
+    "eslint": "eslint .",
+    "flake8": "flake8 .",
+    "lint": "npm run flake8 && npm run eslint"
+  }
+}

--- a/services-marionette/firefox_services_tests/apps/ideatown/app.py
+++ b/services-marionette/firefox_services_tests/apps/ideatown/app.py
@@ -27,4 +27,3 @@ class IdeaTown(Base):
         login_page.sign_up_for_fxa(False, False)
 
         self.wait_for_element_displayed(*self._ideatown_thanks_for_signing_up_header_locator)
-

--- a/services-marionette/firefox_services_tests/apps/pocket/app.py
+++ b/services-marionette/firefox_services_tests/apps/pocket/app.py
@@ -35,4 +35,3 @@ class Pocket(Base):
         with self.marionette.using_context(self.CHROME):
             self.wait_for_element_displayed(
                 *self._pocket_chrome_button).click()
-

